pkg/system-probe/api/client/client.go

```diff
@@ -63,31 +82,87 @@ func get(socketPath string) *http.Client {
 	}
 }
 
-// GetCheck returns data unmarshalled from JSON to T, from the specified module at the /<module>/check endpoint.
-func GetCheck[T any](client *http.Client, module types.ModuleName) (T, error) {
-	checkTelemetry.totalRequests.IncWithTags(map[string]string{checkLabelName: string(module)})
-	var data T
-	req, err := http.NewRequest("GET", ModuleURL(module, "/check"), nil)
-	if err != nil {
-		//we don't have a counter for this case, because this function can't really fail, since ModuleURL function constructs a safe URL
-		return data, err
+func getCheckClient(socketPath string) *CheckClient {
+	return &CheckClient{
+		client:         get(socketPath),
+		startTime:      time.Now(),
+		startupTimeout: pkgconfigsetup.Datadog().GetDuration("check_system_probe_startup_time"),
 	}
+}
 
+func doReq(client *http.Client, req *http.Request, module types.ModuleName) ([]byte, error) {
 	resp, err := client.Do(req)
 	if err != nil {
 		checkTelemetry.failedRequests.IncWithTags(map[string]string{checkLabelName: string(module)})
-		return data, err
+		return nil, err
 	}
 	defer resp.Body.Close()
 
 	body, err := io.ReadAll(resp.Body)
 	if err != nil {
 		checkTelemetry.failedResponses.IncWithTags(map[string]string{checkLabelName: string(module)})
-		return data, err
+		return nil, err
 	}
 	if resp.StatusCode != http.StatusOK {
 		checkTelemetry.responseErrors.IncWithTags(map[string]string{checkLabelName: string(module)})
-		return data, fmt.Errorf("non-ok status code: url %s, status_code: %d, response: `%s`", req.URL, resp.StatusCode, string(body))
+		return nil, fmt.Errorf("non-ok status code: url %s, status_code: %d, response: `%s`", req.URL, resp.StatusCode, string(body))
+	}
+
+	return body, err
+}
+
+func (c *CheckClient) ensureStarted(module types.ModuleName) error {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if c.started {
```

Comment:
minor nit, but it looks like each client is doing its own "started" determination, when we could probably do that for all clients.

Reply:
It would need to properly vary on the socket path though, so maybe not worth the complication.

Reply:
All users with the same socket path are already using a single client thanks to the memoization in GetCheckClient, so this should be already taken care of?